### ISSUE TITLE
Test edges are not excluded from train edges when using RandomLinkSplit.

### ISSUE THREAD

**Issue by panisson** [pyg-team/pytorch_geometric#3440]
## 🐛 Bug

RandomLinkSplit should split the edges in three sets: train, validation and test. Test edges should be excluded from the train set, but in some circumstances they are included.

## To Reproduce

Initialize a Data object as undirected graph. Set the random seed and split into train, validation and test sets using RandomLinkSplit. Depending on the random seed, test edges are included in the train set.

```
import torch
import torch_geometric
from torch_geometric.data import Data
from torch_geometric.transforms import RandomLinkSplit

torch_geometric.seed_everything(42)

edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 3, 4, 4, 5],
                           [1, 0, 2, 1, 3, 2, 4, 3, 5, 4]])

data = Data(edge_index=edge_index, num_nodes=10)
print(data.is_undirected())

transform = RandomLinkSplit(num_val=0.2, num_test=0.2, is_undirected=True, add_negative_train_samples=False)
train_data, val_data, test_data = transform(data)
print(train_data.edge_index)
print(test_data.edge_label_index[:, test_data.edge_label==1])
```

Result:
```
True
tensor([[1, 1, 2, 2, 0, 3],
        [2, 0, 3, 1, 1, 2]])
tensor([[2],
        [1]])
```

## Expected behavior

No edge in the test_data.edge_label_index should be included in the train_data.edge_index, but the edge (1,2) is included in the train edges.
The result depends on the random seed. When initialized with other values, e.g. 1, the problem doesn't happen.

## Environment

* OS: macOS-11.6-x86_64-i386-64bit
* Python version: Python 3.8.5
* PyTorch version: 1.9.1
* CUDA/cuDNN version: no cuda (no GPU)
* GCC version: -
* Any other relevant information: PyTorch Geometric version: 2.0.1

**Comment by rusty1s** [comment]
Thank you so much for letting me know. This is indeed a (pretty silly) bug and I fixed it in master, see https://github.com/pyg-team/pytorch_geometric/commit/213a158b574355c528558a89d531c36b98015a77. I'm sorry for the inconveniences!
